analyzers/src/SonarAnalyzer.Common/Rules/DebuggerDisplayUsesExistingMembersBase.cs

```diff
@@ -0,0 +1,89 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class DebuggerDisplayUsesExistingMembersBase<TAttributeSyntax, TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TAttributeSyntax : SyntaxNode
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S4545";
+
+    private readonly Regex noQuotesModifierExpressionRegex = new(@",\s*nq\s*$", RegexOptions.None, RegexConstants.DefaultTimeout);
+    private readonly Regex evaluatedExpressionRegex = new(@"\{(?<EvaluatedExpression>[^\}]+)\}", RegexOptions.Multiline, RegexConstants.DefaultTimeout);
+
+    protected abstract SyntaxNode GetAttributeFormatString(TAttributeSyntax attribute);
+    protected abstract bool IsValidMemberName(string memberName);
+
+    protected override string MessageFormat => "'{0}' doesn't exist in this context.";
+
+    protected DebuggerDisplayUsesExistingMembersBase() : base(DiagnosticId) { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(Language.GeneratedCodeRecognizer,
+            c =>
+            {
+                var attribute = (TAttributeSyntax)c.Node;
+                if (Language.Syntax.IsKnownAttribute(attribute, KnownType.System_Diagnostics_DebuggerDisplayAttribute, c.SemanticModel)
+                    && GetAttributeFormatString(attribute) is { } formatString
+                    && FirstInvalidMemberName(c, formatString.GetFirstToken().ValueText, attribute) is { } firstInvalidMember)
```

Comment:
The `.GetFirstToken().ValueText` looks unstable. Let's return `string` from `AttributeFormatString` directly. 
That will allow to use `StringValue()` extension and solve some of the interpolation problems
```suggestion
                    && FirstInvalidMemberName(c, formatString, attribute) is { } firstInvalidMember)
```

Reply:
I've kept `SyntaxNode` as return type of `FirstInvalidMemberName`, since I need the node to highlight the location of the format string in 
```cs
c.ReportIssue(Diagnostic.Create(Rule, formatString.GetLocation(), firstInvalidMember));
```

I can still use `StringValue` via `Language.Syntax` at base-class level

```cs
Language.Syntax.StringValue(formatString, c.SemanticModel) is { } formatStringText
```